assets/admin/editor-wizard/editor-wizard-modal.js

```diff
@@ -88,24 +91,41 @@ const EditorWizardModal = () => {
 		} );
 	};
 
-	// TODO Implement different flows depending on post type 👇.
-	const steps = [ CourseDetailsStep, UpgradeStep, CoursePatternsStep ];
-
-	const onWizardCompletion = ( data ) => {
-		// TODO Implement actions when wizard is completed.
-
-		// eslint-disable-next-line no-console
-		console.log(
-			`Wizard completed with data: ${ JSON.stringify( data ) }`
-		);
+	// Choose steps by post type.
+	const stepsByPostType = {
+		course: [ CourseDetailsStep, UpgradeStep, CoursePatternsStep ],
+		lesson: [ LessonDetailsStep, LessonPatternsStep ],
+	};
+	const postType = select( 'core/editor' )?.getCurrentPostType();
+	const steps = stepsByPostType[ postType ];
 
+	// eslint-disable-next-line no-unused-vars
+	const onWizardCompletion = ( wizardData ) => {
+		// TODO Implement actions when wizard is completed
 		closeModal();
 	};
 
+	const updateModalTitle = ( wizardData ) => {
+		if (
+			wizardData.modalTitle !== undefined &&
+			wizardData.modalTitle !== modalTitle
```

Comment:
Technically we don't need this comparison, because React should be able to not re-render everything [if the value set is the same as the previous one](https://reactjs.org/docs/hooks-reference.html#usestate): 

> If you update a State Hook to the same value as the current state, React will bail out without rendering the children or firing effects. (React uses the [Object.is comparison algorithm](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is#Description).)

There's this little issue with this approach, tho, but I'm not sure if this is a huge problem:

> Note that React may still need to render that specific component again before bailing out. That shouldn’t be a concern because React won’t unnecessarily go “deeper” into the tree. If you’re doing expensive calculations while rendering, you can optimize them with useMemo.

So in theory we could simplify the `if` condition to this:

```suggestion
			wizardData.modalTitle
```